_api-reference/grpc-apis/search.md

```diff
@@ -0,0 +1,407 @@
+| query_string | QueryStringQuery | (Optional.) Parses advanced queries from a single string. Must be the only field set. |
+| simple_query_string | SimpleQueryStringQuery | (Optional.) Less strict syntax alternative to `query_string`. Ignores invalid syntax. Must be the only field set. |
+| intervals | map<string, IntervalsQuery> | (Optional.) Matches terms based on position/proximity. Only one entry allowed. Must be the only field set. |
+| knn | map<string, KnnField> | (Optional.) k-Nearest Neighbors query across vector fields. Only one entry allowed. Must be the only field set. |
+| match_all | MatchAllQuery | (Optional.) Matches all documents in the index. Must be the only field set. |
+| match_none | MatchNoneQuery | (Optional.) Matches no documents. Must be the only field set. |
+| script_score | ScriptScoreQuery | (Optional.) Custom scoring using scripts. Must be the only field set. |
+| nested | NestedQuery | (Optional.) Wraps a query targeting nested fields. Must be the only field set. |
+
+## Supported queries 
+
+### 1. Match all query 
+
+Returns all documents from the index, capped to a maximum of 50 docs.
+
+```json
+{
+  "request_body": {
+    "query": {
+      "match_all": {}
+    },
+    "size": 50
+  }
+}
+```
+{% include copy.html %}
+
+### 2. Term query 
+
+Match a single field with a specific term.
+
+```json
+{
+  "index": "my_index",
+  "request_body": {
+    "query": {
+      "term": {
+        "title": {
+          "value": {
+            "string_value": "Rush"
+          },
+          "case_insensitive": true
+        }
+      }
+    }
+  }
+}
+```
+{% include copy.html %}
+
+### 3. Terms query 
+
+Match documents where a specific field contains any  value from a list. 
+
+The following query searches for lines with the IDs 61809 and 61810:
+
+```json
+{
+ "request_body": {
+   "query": {
+     "terms": {
+       "terms_lookup_field_string_array_map": {
+         "line_id": {
+           "string_array": {
+             "string_array": [
+               "61809",
+                    "61810"
+             ]
+           }
+         }
+       }
+     }
+   }
+ }
+}
+EOM
+```
+{% include copy.html %}
+
+### 4. Terms query with terms lookup
+
+Match documents in the `students` index for every student whose `id` matches one of the values in the `enrolled` array:
+
+```json
+{
+  "request_body": {
+    "query": {
+      "terms": {
+        "boost": 1.0,
+        "terms_lookup_field_string_array_map": {
+          "student_id": {
+            "terms_lookup_field": {
+              "index": "classes",
+              "id": "101",
+              "path": "enrolled"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy.html %}
+
+
+### 5. Match none
+
+The counterpart to the Match All query. 
+
+```json
+{
+  "request_body": {
+    "query": {
+      "match_none": {}
+    }
+  }
+}
+```
+{% include copy.html %}
+
+## Response 
+### SearchResponse fields 
+
+| Field | Protobuf Type | Description |
```

Comment:
Please make the format similar to the API template and add an intro sentence for each table.